File: Examples/ButtonsExample/ButtonsView.m
//
//  ButtonsView.m
//  ButtonsExample
//
//  Created by Eric Horacek on 12/27/14.
//  Copyright (c) 2014 Eric Horacek. All rights reserved.
//

#import "ButtonsView.h"

NS_ASSUME_NONNULL_BEGIN

@implementation ButtonsView

#pragma mark - Lifecycle

- (instancetype)initWithFrame:(CGRect)frame {
    self = [super initWithFrame:frame];

    _saveButton = [UIButton buttonWithType:UIButtonTypeSystem];
    _deleteButton = [UIButton buttonWithType:UIButtonTypeSystem];

    [self.deleteButton setTitle:@"Delete" forState:UIControlStateNormal];
    [self.saveButton setTitle:@"Save" forState:UIControlStateNormal];

    NSArray<UIView *> *subviews = @[ self.saveButton, self.deleteButton ];
    UIStackView *stackView = [[UIStackView alloc] initWithArrangedSubviews:subviews];
    stackView.layoutMargins = (UIEdgeInsets){ .left = 10, .right = 10 };
    stackView.layoutMarginsRelativeArrangement = YES;
    stackView.spacing = 10;
    stackView.distribution = UIStackViewDistributionFillEqually;
    [self addSubview:stackView];

    stackView.translatesAutoresizingMaskIntoConstraints = NO;
    [stackView.centerYAnchor constraintEqualToAnchor:self.centerYAnchor].active = YES;
    [stackView.centerXAnchor constraintEqualToAnchor:self.centerXAnchor].active = YES;
    [stackView.widthAnchor constraintEqualToAnchor:self.widthAnchor].active = YES;

    return self;
}

#pragma mark - UIView

+ (BOOL)requiresConstraintBasedLayout {
    return YES;
}

@end

NS_ASSUME_NONNULL_END

File: Examples/SwiftButtonsExample/ButtonsView.swift
//
//  ButtonsView.swift
//  SwiftButtonsExample
//
//  Created by Eric Horacek on 3/15/15.
//  Copyright (c) 2015 Eric Horacek. All rights reserved.
//

import UIKit

class ButtonsView: UIView {

    // MARK: - Lifecycle

    override init(frame: CGRect) {
        saveButton = UIButton(type: .system)
        deleteButton = UIButton(type: .system)

        saveButton.setTitle("Save", for: UIControlState())
        deleteButton.setTitle("Delete", for: UIControlState())

        super.init(frame: frame)

        let stackView = UIStackView(arrangedSubviews: [ saveButton, deleteButton ])
        stackView.layoutMargins = UIEdgeInsetsMake(0, 10, 0, 10)
        stackView.isLayoutMarginsRelativeArrangement = true
        stackView.spacing = 10
        stackView.distribution = .fillEqually
        addSubview(stackView)

        stackView.translatesAutoresizingMaskIntoConstraints = false
        stackView.centerYAnchor.constraint(equalTo: self.centerYAnchor).isActive = true
        stackView.centerXAnchor.constraint(equalTo: self.centerXAnchor).isActive = true
        stackView.widthAnchor.constraint(equalTo: self.widthAnchor).isActive = true
    }

    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }

    // MARK: - UIView

    override class var requiresConstraintBasedLayout : Bool {
        return true
    }

    // MARK: - ButtonsView
    
    let saveButton: UIButton
    let deleteButton: UIButton
}
